src/ArrayObject.php

```diff
@@ -161,6 +173,9 @@ public function &__get(mixed $key)
     /**
      * Appends the value
      *
+     * @template TInputValue
+     * @param TInputValue $value
+     * @psalm-self-out ArrayObject<TKey|int, TValue|TInputValue>
```

Comment:
`TKey` is `array-key` (`string|int`) so the `|int` shouldn't be necessary

Reply:
This is relevant if `TKey` was only `string`

Reply:
In practice, an `ArrayObject<string, mixed>` becomes an `ArrayObject<string|int, mixed>` here, when calling this method